src/Features/CSharp/Portable/DocumentationComments/DocumentationCommentSnippetService.cs

```diff
@@ -111,22 +112,49 @@ protected override List<string> GetDocumentationCommentStubLines(MemberDeclarati
                 }
             }
 
-            if (member.IsKind(SyntaxKind.MethodDeclaration) ||
-                member.IsKind(SyntaxKind.IndexerDeclaration) ||
-                member.IsKind(SyntaxKind.DelegateDeclaration) ||
-                member.IsKind(SyntaxKind.OperatorDeclaration))
+            if (member.IsKind(
+                    SyntaxKind.MethodDeclaration,
+                    SyntaxKind.IndexerDeclaration,
+                    SyntaxKind.DelegateDeclaration,
+                    SyntaxKind.OperatorDeclaration,
+                    SyntaxKind.ConstructorDeclaration,
+                    SyntaxKind.DestructorDeclaration))
             {
                 var returnType = member.GetMemberType();
                 if (returnType != null &&
                     !(returnType.IsKind(SyntaxKind.PredefinedType, out PredefinedTypeSyntax? predefinedType) && predefinedType.Keyword.IsKindOrHasMatchingText(SyntaxKind.VoidKeyword)))
                 {
                     list.Add("/// <returns></returns>");
                 }
+
+                foreach (var exceptionType in GetExceptions(member))
+                {
+                    list.Add(@$"/// <exception cref=""{exceptionType}""></exception>");
+                }
             }
 
             return list;
         }
 
+        private static IEnumerable<string> GetExceptions(SyntaxNode member)
+        {
+            var statements = member.DescendantNodes().Where(n => n.IsKind(SyntaxKind.ThrowExpression, SyntaxKind.ThrowStatement));
+            foreach (var statement in statements)
+            {
+                var expression = statement switch
+                {
+                    ThrowExpressionSyntax throwExpression => throwExpression.Expression,
+                    ThrowStatementSyntax throwStatement => throwStatement.Expression,
+                    _ => throw ExceptionUtilities.Unreachable
+                };
+
+                if (expression is ObjectCreationExpressionSyntax { Type: TypeSyntax type })
+                {
+                    yield return type.ToString();
```

Comment:
Likely won't work with generics. But that's likely so niche as to not be relevant. 

Is also like it if we filtered it exceptions in an inner scope that are caught before escaping.